app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,73 @@
+class MicropostsController < ApplicationController
+  before_action :set_micropost, only: %i[ show edit update destroy ]
+
+  # GET /microposts or /microposts.json
+  def index
+    @microposts = Micropost.all
+  end
+
+  # GET /microposts/1 or /microposts/1.json
+  def show
+  end
+
+  # GET /microposts/new
+  def new
+    @micropost = Micropost.new
+  end
+
+  # GET /microposts/1/edit
+  def edit
+  end
+
+  # POST /microposts or /microposts.json
+  def create
+    @micropost = Micropost.new(micropost_params)
+
+    respond_to do |format|
+      if @micropost.save
+        format.html { redirect_to @micropost, notice: "Micropost was successfully created." }
+        format.json { render :show, status: :created, location: @micropost }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @micropost.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /microposts/1 or /microposts/1.json
+  def update
+    respond_to do |format|
+      if @micropost.update(micropost_params)
+        format.html { redirect_to @micropost, notice: "Micropost was successfully updated." }
+        format.json { render :show, status: :ok, location: @micropost }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @micropost.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /microposts/1 or /microposts/1.json
+  def destroy
+    respond_to do |format|
+      if @micropost.destroy(micropost_params)
```

Comment:
```suggestion
      if @micropost.destroy
```

Read through the documentation on how the `destroy` works. Also, make sure you understand the difference between `destroy` vs `destroy!`

Reply:
https://apidock.com/rails/v5.2.3/ActiveRecord/Persistence/destroy

https://apidock.com/rails/v5.2.3/ActiveRecord/Persistence/destroy%21